File: wrappers/mvc/src/Kendo.Mvc/UI/JsonObject.cs
using System.Collections.Generic;

namespace Kendo.Mvc.UI
{
    public abstract class JsonObject
    {
        public IDictionary<string, object> ToJson()
        {
            var json = new Dictionary<string, object>();

            Serialize(json);

            return json;
        }

        protected abstract void Serialize(IDictionary<string, object> json);
    }
}

File: wrappers/php/lib/kendo/JsonObject.php
<?php

namespace kendo;

abstract class JsonObject implements Serializable {
    private $properties = array();

    public function toJSON() {
        $serializer = new Serializer();

        return $serializer->serialize($this);
    }

    public function properties() {
        return $this->properties;
    }

    protected function setProperty($key, $value) {
        if ($value instanceof Serializable) {
            $value = $value->properties();
        }

        $this->properties[$key] = $value;
    }
}

?>
